### PR TITLE
Update README with overview and workflow

### PR DESCRIPTION
**Pull request by dperny**
Updates the README with an overview of the testing philosophy and the
recommended development workflow. Relies on PRs #22 and #23.

Signed-off-by: Drew Erny <drew.erny@docker.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,39 +1,69 @@
-docker-e2e
-==========
+# docker-e2e #
```
Comment (dongluochen) [inline README.md]: `# docker-e2e` would do. The second `#` is redundant.

File: README.md [hunk 2]
```diff
@@ -1,39 +1,69 @@
...
 tests, the goal of `docker-e2e` is to issue commands strictly through public 
 apis to real Docker Engines running real containers. 
 
-Building the Tests
-------------------
-The tests require go version 1.7+
 
-To compile the tests to a static binary, use `go test -c`.
+## Overview ##
+The tests are built with the `go test` framework, and in some circumstances,
```
Comment (dongluochen) [inline README.md]: `in some circumstances` -> should be more clear about this. We should focus on major scenario first.

File: README.md [hunk 3]
```diff
@@ -1,39 +1,69 @@
...
+The tests are designed to be run inside of a docker container, which should be
+started with `--net=host` and the docker socket bind-mounted into the container
+at `/var/run/docker.sock`. This container should be run on a Manager node, or
+the API calls related to cluster management will all fail.
 
-Alternatively, tests can be run in a docker container. You will have to mount
-`/var/run/docker.sock` into the container, as well as supplying an ip address
-that the container can hit to check network functionality. The endpoint is
-passed as an environment variable, `DOCKER_E2E_ENDPOINT`. 
+The tests must be run on a Docker Swarm Mode manager node.
```
Comment (dongluochen) [inline README.md]: The previous paragraph already includes this.

File: README.md [hunk 4]
```diff
@@ -1,39 +1,69 @@
...
+
+## Development Workflow ##
+
+Before you start working, make sure you have a development cluster ready to go.
+If you're using `testkit`, you can do `testkit create --name somename <cfg>`
```
Comment (dongluochen) [inline README.md]: Is `testkit` optional or not?

File: README.md [hunk 5]
```diff
@@ -1,39 +1,69 @@
...
 
-The tests must be run on a Docker Swarm Mode manager node.
+This first builds and then runs the tests on the remote cluster. If you need
+and ssh connection to the remote cluster to debug something, you can do 
+`testkit ssh somename` to get an IP address that you can use as an SSH 
+endpoint. When you're done with your cluster, you should do 
+`testkit rm somename` to delete it.
 
 Running the Tests with Testkit
 ------------------------------
```
Comment (dongluochen) [inline README.md]: Use `#` style to be consistent.

**Comment by dperny** [comment]
@dongluochen fixed readme nits. should be better now.

**Comment by dongluochen** [comment]
LGTM

File: README.md [hunk 6]
```diff
@@ -1,39 +1,64 @@
...
-Running the Tests with Testkit
-------------------------------
+If you are not using testkit, you can set this socket up manually, or you can 
+simply ssh into the machine and copy over your working code. 
+
+Finally, you're ready to build and run the tests. Do: 
+
+```
+$ docker build -t e2e .
+$ docker run --net=host -v /var/run/docker.sock:/var/run/docker.sock
```
Comment (dmcgowan) [inline README.md]: define an image name or set to the build image id in the command

**Comment by dmcgowan** [comment]
LGTM

**Comment by aluzzardi** [comment]
LGTM

**Pull request merged**